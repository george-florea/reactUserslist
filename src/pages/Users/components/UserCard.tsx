import {
  Card,
  CardBody,
  Stack,
  Heading,
  Divider,
  CardFooter,
  ButtonGroup,
  Button,
  Image,
  Text,
} from "@chakra-ui/react";
import React from "react";

interface IUserCardProps {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  username: string;
  image: string;
}
const UserCard = ({
  id,
  firstName,
  lastName,
  email,
  username,
  image,
}: IUserCardProps) => {
  return (
    <Card data-testid="user">
      <CardBody>
        <Image margin={"auto"} src={image} alt={email} borderRadius="lg" />
        <Stack mt="6" spacing="3">
          <Heading color="blue.600" size="md" textAlign={"center"}>
            @{username}
          </Heading>
          <Stack flexDir="row">
            <Text fontWeight={"800"}>First name:</Text>
            <Text>{firstName}</Text>
          </Stack>
          <Stack flexDir="row">
            <Text fontWeight={"800"}>Last name:</Text>
            <Text>{lastName}</Text>
          </Stack>
          <Text>{email}</Text>
        </Stack>
      </CardBody>
    </Card>
  );
};

export default UserCard;
