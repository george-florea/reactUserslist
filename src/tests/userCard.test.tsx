import React from "react";
import { render, screen } from "@testing-library/react";
import UserCard from "../pages/Users/components/UserCard";
import "@testing-library/jest-dom";

describe("UserCard Component", () => {
  const mockProps = {
    id: 1,
    firstName: "John",
    lastName: "Doe",
    email: "john.doe@example.com",
    username: "johndoe",
    image: "https://example.com/image.jpg",
  };

  test("renders UserCard with correct information", () => {
    render(<UserCard {...mockProps} />);

    const image = screen.getByRole("img");
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute("src", mockProps.image);

    const userNameHeading = screen.getByText(`@${mockProps.username}`);
    expect(userNameHeading).toBeInTheDocument();

    expect(screen.getByText("First name:")).toBeInTheDocument();
    expect(screen.getByText(mockProps.firstName)).toBeInTheDocument();

    expect(screen.getByText("Last name:")).toBeInTheDocument();
    expect(screen.getByText(mockProps.lastName)).toBeInTheDocument();

    expect(screen.getByText(mockProps.email)).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<UserCard {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
