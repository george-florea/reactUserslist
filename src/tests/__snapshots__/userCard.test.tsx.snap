// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="chakra-card css-2dlz7"
    data-testid="user"
  >
    <div
      class="chakra-card__body css-0"
    >
      <img
        alt="john.doe@example.com"
        class="chakra-image css-gnhd0a"
        src="https://example.com/image.jpg"
      />
      <div
        class="chakra-stack css-29dqc6"
      >
        <h2
          class="chakra-heading css-1e61p8h"
        >
          @johndoe
        </h2>
        <div
          class="chakra-stack css-1f0wxn3"
        >
          <p
            class="chakra-text css-ty7r4z"
          >
            First name:
          </p>
          <p
            class="chakra-text css-0"
          >
            John
          </p>
        </div>
        <div
          class="chakra-stack css-1f0wxn3"
        >
          <p
            class="chakra-text css-ty7r4z"
          >
            Last name:
          </p>
          <p
            class="chakra-text css-0"
          >
            Doe
          </p>
        </div>
        <p
          class="chakra-text css-0"
        >
          john.doe@example.com
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
